provider/spark_configs.go

```diff
@@ -349,6 +429,119 @@ func (flag sparkDeployFlag) SparkFlags() sparkFlags {
 	}
 }
 
+type sparkSnowflakeFlags struct {
+	Config       *pc.SnowflakeConfig
+	ExecutorType pc.SparkExecutorType
+}
+
+func (args sparkSnowflakeFlags) SparkFlags() sparkFlags {
+	if args.Config == nil {
+		logging.GlobalLogger.Debug(
+			"Not setting spark snowflake flags, snowflake config not set",
+		)
+		return sparkFlags{}
+	}
+	flags := sparkFlags{
+		sparkCredFlag{
+			Key:   "sfURL",
+			Value: args.Config.GetBaseURL(),
+		},
+		sparkCredFlag{
+			Key:   "sfUser",
+			Value: args.Config.Username,
+		},
+		sparkCredFlag{
+			Key:   "sfPassword",
+			Value: args.Config.Password,
+		},
+		sparkCredFlag{
+			Key:   "sfWarehouse",
+			Value: args.Config.Warehouse,
+		},
+	}
+	// Databricks has these packages pre-installed.
+	if args.ExecutorType == pc.EMR {
+		flags = append(flags, sparkPackagesFlag{
+			Packages: []string{
+				"net.snowflake:snowflake-jdbc:3.13.22",
+				"net.snowflake:spark-snowflake_2.12:2.12.0-spark_3.4",
+			},
+		})
+	}
+	return flags
+}
+
+// This is based on very legacy values and aren't tested
+type sparkAzureFlags struct {
+	AccountName      string
+	AccountKey       string
+	ConnectionString string
+	ContainerName    string
+}
+
+func (args sparkAzureFlags) SparkFlags() sparkFlags {
+	return sparkFlags{
+		sparkConfigFlag{
+			Key:   fmt.Sprintf("fs.azure.account.key.%s.dfs.core.windows.net", args.AccountName),
+			Value: args.AccountKey,
+		},
+		sparkCredFlag{
+			Key:   "azure_connection_string",
+			Value: args.ConnectionString,
+		},
+		sparkCredFlag{
+			Key:   "azure_container_name",
+			Value: args.ContainerName,
+		},
+		sparkPackagesFlag{
+			Packages: []string{"org.apache.hadoop:hadoop-azure-3.2.0"},
```

Comment:
Same concern here: we might want to throw these in `config` with the current values as defaults.